lua/telekasten.lua

```diff
@@ -197,10 +197,52 @@ local monthmap = {
     "December",
 }
 
+local dateformats = {
+    date = "%Y-%m-%d",
+    week = "%V",
+    isoweek = "%Y-W%V"
+}
+
+local function relativedates_today()
+    local now = os.time()
+    local dinfo = os.date("*t")
+    local wday = dinfo.wday - 1
+    local oneday = 24 * 60 * 60 -- hours * days * seconds
+    local oneweek = 7 * oneday
+    local df = dateformats
+    local opts = {}
+
+    opts.yesterday   = os.date(df.date, now - oneday)
+    opts.tomorrow    = os.date(df.date, now + oneday)
+    opts.lastweek    = os.date(df.week, now - oneweek)
+    opts.nextweek    = os.date(df.week, now + oneweek)
+    opts.isolastweek = os.date(df.isoweek, os.time() - oneweek)
+    opts.isonextweek = os.date(df.isoweek, os.time() + oneweek)
+
+    -- Find the Sunday that started this week regardless of the calendar
+    -- display preference.  Then use that as the base to calculate the dates
+    -- for the days of the current week.
+    -- Finally, adjust Sunday to suit user calendar preference.
+    local starting_sunday = now - (wday * oneday)
+    local sunday_offset = 0
+    if M.Cfg.calendar_opts.calendar_monday == 1 then
+      sunday_offset = 7
+    end
+    opts.monday    = os.date(df.date, starting_sunday + (1 * oneday))
+    opts.tuesday   = os.date(df.date, starting_sunday + (2 * oneday))
+    opts.wednesday = os.date(df.date, starting_sunday + (3 * oneday)) 
+    opts.thursday  = os.date(df.date, starting_sunday + (4 * oneday))
+    opts.friday    = os.date(df.date, starting_sunday + (5 * oneday))
+    opts.saturday  = os.date(df.date, starting_sunday + (6 * oneday))
+    opts.sunday    = os.date(df.date, starting_sunday + (sunday_offset * oneday))
+
+    return opts
+end
+
 local function calenderinfo_today()
     local dinfo = os.date("*t")
```

Comment:
I strongly recommend merging `relativedates_today()` or its output into the returned `opts`. This becomes apparent when I address the `linesubst()` function.